tfhe/src/integer/server_key/radix_parallel/vector_find.rs

```diff
@@ -0,0 +1,1282 @@
+use crate::core_crypto::prelude::UnsignedInteger;
+use crate::integer::block_decomposition::{BlockDecomposer, Decomposable, DecomposableInto};
+use crate::integer::{BooleanBlock, IntegerRadixCiphertext, RadixCiphertext, ServerKey};
+use crate::prelude::CastInto;
+use crate::shortint::Ciphertext;
+use itertools::Itertools;
+use rayon::prelude::*;
+use std::hash::Hash;
+use std::usize;
+
+impl ServerKey {
+    /// `match` an input value to an output value
+    ///
+    /// - Input values _must_ be unique.
```

Comment:
I'm ok to clean luts with non unique values since it can be done in clear so performances won't be affected as much

However what do we do if the lut has a non unique input values ? only keep the first/last/random ? output and error ?

Reply:
I was thinking like taking an input array or function and basically hashing it to keep unique values in the Lut object wdyt ?

Reply:
Not sure I understand, If I only have an input array I can't do anything because I also need the outputs

I still want to keep an entry point that works with an already built lut, because passing a function that builds a lut from an already existing lut feels a bit stupid (e.g AES 128 gives a lut, so if we only accept function, one would need to create a function that will look into the lut and return the value the lut gives him)

We could have a 
```
struct MatchValue<Clear> { input: Clear, output: Clear };
struct MatchValues<Clear> (Vec<MatchValue<Clear>);

impl<Clear> MatchValues<Clear> {
    fn from_function<F: Fn(Clear) -> Clear>>(fn: F, input_set: &[Clears]) -> Self ;
}
```

But to apply a function over the whol range of values possible for the input ciphertext, I have a local implementation which does not require allocating a LUT allowing to save some memory

Reply:
Ok what we should do is have a `MatchValues` types (or some other) names that has different constructors (from already existing luts, input and output array, input array and function, and these constructors would ensure the uniqueness of inputs in the final object, reporting errors if necessary ?

Reply:
Yes